Providing a path to the custom plugin in the 'import' field of the allurerc config will throw an error trying to generate a report. This is a regression after upgrading from 3.3.1.

The is issue is in https://github.com/allure-framework/allure3/blame/7563796608a2e37d04c570b7f914d8af7b40dd60/packages/core/src/config.ts#L400

const isModuleNotFoundError = (err: unknown): err is Error & { code: "ERR_MODULE_NOT_FOUND" } => {
  return err instanceof Error && "code" in err && err.code === "ERR_MODULE_NOT_FOUND";
};

which checks for "ERR_MODULE_NOT_FOUND" instead of "MODULE_NOT_FOUND" and therefore not proceeding and throwing the error.

**To Reproduce**
Steps to reproduce the behavior:
1. Add a custom plugin to the allurerc.mjs file, for example:
plugins: {
    "custom-report": {
      import: "/Users/path_to_your_plugin/packages/plugin-awesome/dist/index.js"),
2. Generate a report: npx allure generate
3. Error:
Internal Error: Cannot find module '@allurereport/plugin-/Users/path_to_your_plugin/custom-report/packages/plugin-custom-report/dist/index.js'

**Expected behavior**
A successful generation of the report

--- a/packages/core/src/config.ts
+++ b/packages/core/src/config.ts
@@ -396,8 +396,10 @@ export const getPluginInstance = (config: FullConfig, predicate: (plugin: Plugin
  *
  * @see https://nodejs.org/api/errors.html#err-module-not-found
  */
-const isModuleNotFoundError = (err: unknown): err is Error & { code: "ERR_MODULE_NOT_FOUND" } => {
-  return err instanceof Error && "code" in err && err.code === "ERR_MODULE_NOT_FOUND";
+const isModuleNotFoundError = (err: unknown): err is Error & { code: "ERR_MODULE_NOT_FOUND" | "MODULE_NOT_FOUND" } => {
+  return (
+    err instanceof Error && "code" in err && (err.code === "ERR_MODULE_NOT_FOUND" || err.code === "MODULE_NOT_FOUND")
+  );
 };
 
 export const resolvePlugin = async (path: string) => {

--- a/packages/core/test/config.test.ts
+++ b/packages/core/test/config.test.ts
@@ -227,40 +227,52 @@ describe("getPluginId", () => {
   });
 });
 
-class ModuleNotFoundError extends Error {
-  constructor() {
-    super("Module not found");
-  }
+const createModuleNotFoundError = (code: "ERR_MODULE_NOT_FOUND" | "MODULE_NOT_FOUND") => {
+  const err = new Error(code === "MODULE_NOT_FOUND" ? "Cannot find module" : "Module not found");
 
-  code = "ERR_MODULE_NOT_FOUND";
-}
+  return Object.assign(err, { code });
+};
 
 describe("resolvePlugin", () => {
+  const customPluginPath = "/tmp/custom-plugin/index.js";
+
   beforeEach(() => {
     vi.clearAllMocks();
   });
 
-  it("prepends @allurereport/plugin- prefix and tries to resolve plugin when the path is not scoped", async () => {
+  it.each([
+    {
+      title: "prepends @allurereport/plugin- prefix and falls back on ESM module-not-found",
+      pluginPath: "classic",
+      code: "ERR_MODULE_NOT_FOUND" as const,
+    },
+    {
+      title: "falls back to the original path on CommonJS module-not-found",
+      pluginPath: customPluginPath,
+      code: "MODULE_NOT_FOUND" as const,
+    },
+  ])("$title", async ({ pluginPath, code }) => {
     const fixture = { name: "Allure" };
+    const expectedPrefixedPath = `@allurereport/plugin-${pluginPath}`;
 
     (importWrapper as unknown as MockInstance).mockImplementation((path: string) => {
-      if (path.startsWith("@allurereport")) {
-        throw new ModuleNotFoundError();
+      if (path === expectedPrefixedPath) {
+        throw createModuleNotFoundError(code);
       }
 
       return { default: fixture };
     });
 
-    const plugin = await resolvePlugin("classic");
+    const plugin = await resolvePlugin(pluginPath);
 
     expect(importWrapper).toHaveBeenCalledTimes(2);
-    expect(importWrapper).toHaveBeenCalledWith("@allurereport/plugin-classic");
-    expect(importWrapper).toHaveBeenCalledWith("classic");
+    expect(importWrapper).toHaveBeenNthCalledWith(1, expectedPrefixedPath);
+    expect(importWrapper).toHaveBeenNthCalledWith(2, pluginPath);
     expect(plugin).toEqual(fixture);
   });
 
   it("throws an error when plugin can't be resolved", async () => {
-    (importWrapper as unknown as MockInstance).mockRejectedValue(new ModuleNotFoundError());
+    (importWrapper as unknown as MockInstance).mockRejectedValue(createModuleNotFoundError("ERR_MODULE_NOT_FOUND"));
 
     await expect(() => resolvePlugin("classic")).rejects.toThrow("Cannot resolve plugin: classic");
   });

